### PR TITLE
Unify logging

### PR DESCRIPTION
**Pull request by linaScience**

### CODE REVIEW

File: src/EventLoop.cpp [hunk 1]
```diff
@@ -28,20 +28,21 @@ void EventLoop::loop_ipc_queue(const std::future<void>& future) {
...
+        WMLOG(ERROR) << "Could not connect station: " << event->station.mac.string()
+                     << " with VNI: " << event->station.vni() << " to VXLAN interface. " << err.what()
+                     << " - Will now send DEAUTH packet.";
         caller.deauth_station(event->station);
     }
 }
 
 void EventLoop::handle_disconnect(ipc::DisconnectEvent* event) {
-    WMLOG(INFO) << "Station " << event->station << " disconnected from AP";
+    WMLOG(INFO) << "Disconnected station: " << event->station.mac.string();
```
Comment (Sohn123) [inline src/EventLoop.cpp:47]: This sounds like we disconnected the station but rather the station decided to disconnect.

File: src/ipc/Socket.cpp [hunk 2]
```diff
@@ -74,29 +74,29 @@ ipc::Socket::Socket(const std::chrono::duration<int>& timeout, const std::string
     local.sun_path[sizeof(local.sun_path) - 1] = '\0';
     if (bind(sock_fd, reinterpret_cast<struct sockaddr*>(&local), sizeof(local)) == -1) {
         close(sock_fd);
-        throw std::runtime_error("could not bind to socket " + dest_path + ": " + std::strerror(errno));
+        throw std::runtime_error("Could not bind to socket " + dest_path + ": " + std::strerror(errno));
```
Comment (Sohn123) [inline src/ipc/Socket.cpp:77]: Colon missing.

File: src/ipc/Socket.cpp [hunk 3]
```diff
@@ -118,7 +118,7 @@ void ipc::Socket::send_command(const std::vector<std::string>& args) {
 
     ssize_t err = send(sock_fd, command.c_str(), command.size(), 0);
     if (err < 0) {
-        throw std::runtime_error(std::string("could not send_command to socket: ") + std::strerror(errno));
+        throw std::runtime_error(std::string("Could not send_command() to socket: ") + std::strerror(errno));
```
Comment (Sohn123) [inline src/ipc/Socket.cpp:121]: I would write `Could not send to socket: `

File: src/nl/Socket.cpp [hunk 4]
```diff
@@ -122,7 +122,7 @@ void nl::Socket::delete_interface(const std::string &name) {
     if (err < 0) {
         throw std::runtime_error(std::string("Could not delete link: ") + nl_geterror(err));
     }
-    WMLOG(INFO) << "Deleted " << name;
+    WMLOG(INFO) << "Deleted interface." << name;
```
Comment (Sohn123) [inline src/nl/Socket.cpp:125]: Colon instead of dot

**Review by Sohn123 — CHANGES_REQUESTED** [review]
I think the format you developed is good :) Just some minor corrections and nitpicks. So let's fix it and :shipit:

**Comment by Sohn123** [comment]
Please also rebase onto latest main so the pipeline is green.

File: src/Configuration.cpp [hunk 5]
```diff
@@ -44,9 +44,9 @@ void Configuration::set_all_available_sockets_if_empty() {
             }
         }
         if (socknames.empty()) {
-            throw std::runtime_error("No sockets were configured and no sockets could be found.");
+            throw std::runtime_error("Could not find and configure sockets.");
         }
-        WMLOG(DEBUG) << "No sockets were configured. Using all " << socknames.size() << " sockets available.";
+        WMLOG(DEBUG) << "Could not configure sockets. Using all " << socknames.size() << " sockets available.";
```
[the remark this answers is not in the document]
Comment (linaScience) [reply src/Configuration.cpp:49]: Does `"Found" << socknames.size() << "available sockets".` sound better?

File: src/BridgePerVxlanRenderer.cpp [hunk 6]
```diff
@@ -11,24 +11,24 @@ BridgePerVxlanRenderer::BridgePerVxlanRenderer()
...
+    WMLOG(DEBUG) << "Calling setup_station() with station:" << station;
 
     uint32_t vni = station.vni();
     socket.create_vxlan_iface(vni);
     socket.create_bridge_for_vni(vni);
-    socket.add_iface_bridge("bridge" + std::to_string(vni), "vxlan" + std::to_string(vni));
 
     if (not station.vlan_id.has_value()) {
-        throw std::runtime_error("The station " + station.mac.string() + " has no vlan_id");
+        throw std::runtime_error("Station with MAC: " + station.mac.string() + " has no vlan_id.");
```
Comment (rgwohlbold) [inline src/BridgePerVxlanRenderer.cpp:21]: In my opinion, we should use key-value pairs to give additional information, and move them to the end.
For this log message, I propose "Station has no vlan_id mac=<mac>", similar for the others

File: src/nl/Socket.cpp [hunk 7]
```diff
@@ -122,7 +122,7 @@ void nl::Socket::delete_interface(const std::string &name) {
     if (err < 0) {
         throw std::runtime_error(std::string("Could not delete link: ") + nl_geterror(err));
     }
-    WMLOG(INFO) << "Deleted " << name;
+    WMLOG(INFO) << "Deleted interface:" << name;
```
Comment (rgwohlbold) [inline src/nl/Socket.cpp:125]: use `iface=`

File: src/nl/Socket.cpp [hunk 8]
```diff
@@ -91,28 +91,28 @@ void nl::Socket::add_iface_bridge(const std::string &bridge_name, const std::str
...
 
     err = rtnl_link_enslave(socket, bridge_link.link, interface_link.link);
     if (err < 0) {
-        throw std::runtime_error("Could not enslave interface: " + interface_name + " to Bridge " + bridge_name + " " +
-                                 nl_geterror(err));
+        throw std::runtime_error("Could not enslave interface= " + interface_name + " to bridge= " + bridge_name +
+                                 " : " + nl_geterror(err));
     }
-    WMLOG(INFO) << "Connected " << interface_name << " to " << bridge_name;
+    WMLOG(INFO) << "Connected interface=" << interface_name << " to bridge= " << bridge_name;
```
Comment (rgwohlbold) [inline src/nl/Socket.cpp:109]: We should put the KV-pairs last

**Review by rgwohlbold — CHANGES_REQUESTED** [review]
Our goal here is to make our logs searchable.
For example, if I wanted to find all events relating to a  MAC address `00:00:00:00:00:01`, I would search for `station=00:00:00:00:00:01`.
For that reason, I propose the following changes:

- We should use unified keys, for example `station` instead of `station` or `mac`
- No whitespace between the `=` and the value
- We should put the key-value pairs last and leave out prepositions like `with` before the pairs.

File: src/nl/Socket.cpp (hunk 8 above)
Comment (linaScience) [reply src/nl/Socket.cpp:109]: Would `"Connected iface=" << interface_name << " bridge=" << bridge_name;` fit here?
Or do you suggest something else for this case?

**Review by Sohn123 — APPROVED** [review]
For me it is fine now. Thanks for all the work you put in @linaScience What do you think @rgwohlbold ?

**Review by rgwohlbold — APPROVED** [review]
LGTM :shipit:

**Pull request merged**